doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1265 @@
+  
+
+sonic# show alarm detail 
+  
+alarm details - 14
+-------------------------------------------
+Id:                14
+Severity:          CRITICAL
+Source:            /sensor/2
+Name:              TEMPERATURE_EXCEEDED
+Description:       temperatureCrossedThreshold: Current temperature of sensor/2 is 76 degrees
+Raise-time:        Wed Feb 10 18:08:24 2021
+Ack-time:          
+New:               true
+Acknowledged:      false
+
+sonic# show alarm from 14 to 16
+----------------------------------------------------------------------------------------------------------------------------
+Id           Severity   Name                           Timestamp                   Description                                                  
+----------------------------------------------------------------------------------------------------------------------------
+14           WARNING    TEMPERATURE_EXCEEDED           2021-05-20T00:47:52.992Z    :- temperatureCrossedThreshold: Current temperature of sensor/2 is 76 degrees         
+15           WARNING    DUMMY_ALARM                    2021-05-20T02:16:41.637Z    :- signalHandler: Raising simulated alarm         
+16           WARNING    PSU_FAULT                      2021-05-20T02:16:42.611Z    :- /psu/2 has experienced a fault 
+
+sonic# show alarm summary
+Alarm summary
+---------------------------------
+Total:                         3 
+Critical:                      0 
+Major:                         0 
+Minor:                         0 
+Warning:                       3 
+Acnowledged:                   2 
+----------------------------------
+```
+
+### 3.3.3 REST API Support
+
+sonic REST links:
+*  /restconf/data/sonic-event:sonic-event/EVENT/EVENT_LIST
+*  /restconf/data/sonic-event:sonic-event/EVENT_STATS/EVENT_STATS_LIST
+*  /restconf/data/sonic-alarm:sonic-alarm/ALARM/ALARM_LIST
+*  /restconf/data/sonic-alarm:sonic-alarm/ALARM_STATS/ALARM_STATS_LIST
+*  /restconf/operations/sonic-evprofile:get-evprofile
+*  /restconf/operations/sonic-evprofile:set-evprofile
+*  /restconf/operations/sonic-alarm:acknowledge-alarms
+*  /restconf/operations/sonic-alarm:unacknowledge-alarms
+
+openconfig REST links:
+*  /restconf/data/openconfig-system:system/openconfig-events:events
+*  /restconf/data/openconfig-system:system/openconfig-events:event-stats
+*  /restconf/data/openconfig-system:system/alarms
+*  /restconf/data/openconfig-system:system/openconfig-alarms-ext:alarm-stats
+
+# 4 Flow Diagrams
+![Sequence Diagram](event-alarm-framework-seqdiag.png)
+
+# 5 Warm Boot Support
+## 5.1 Application warm boot
+Applications confirming to the warm boot, should have stored their state and compare current values against previous values.
+Such compliant application also "remembers" that it raised an event before for a specific condition.
+They would 
+*  not raise alarms/events for the same condition that it raised pre warm boot
```

Comment:
I believe this is hard req for application and logic that should not resides in the application but in the infra. 
@lguohan, @yxieca what do you think?

Reply:
@lguohan @yxieca  Please comment and help bring this to a conclusion.